lib/hci-socket/gatt.js

```diff
@@ -150,7 +150,7 @@ Gatt.prototype.onAclStreamData = function(cid, data) {
 };
 
 Gatt.prototype.onAclStreamEncrypt = function(encrypt) {
-  if (encrypt) {
+  if (encrypt && this._currentCommand) {
```

Comment:
Can you please provide an HCI dump for when this case is triggered? (and description as well).

The changes in `lib/hci-socket/smp.js` look good to me!